### PR TITLE
Feature api endpoints

### PR DESCRIPTION
**Pull request by Graciano1997**
In this milestone we did the bellow project's requirements:

- Create an API endpoint to list all posts for a user.
- Create an API endpoint to list all comments for a user's post.
- Create an API endpoint to add a comment to a post. Remember that the owner of the comment is the user that makes it; it should not be set by passing a parameter.
- Your API endpoints should receive JSON and respond JSON as well.
- Add JWT for generate token
Update readme
fix linter issues

### CODE REVIEW

File: app/controllers/API/V1/comments_controller.rb [hunk 1]
```diff
@@ -9,6 +10,7 @@ def index
   end
 
   def create
+    authorize! :create, Comment
     @post = Post.find(params[:post_id])
     @comment = @post.comments.new(text: comment_params[:text], user_id: current_user.id)
```
Comment (danmainah) [inline app/controllers/API/V1/comments_controller.rb:15]: - [ ] @Team Great work adding controllers for your API, when I  run the API in Postman I get the error below
image
image

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @Graciano1997 and @pushkar1995 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ Added API endpoints to list posts, comments and add comments.
✔️ Added route to access the endpoints.

# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by Graciano1997** [comment]
Hello Code reviewers 

image


image


image

image
☝️ ☝️  shows the user token message

File: app/controllers/API/V1/posts_controller.rb [hunk 2]
```diff
@@ -1,7 +1,11 @@
 class Api::V1::PostsController < ApplicationController
   load_and_authorize_resource
+  skip_before_action :authenticate_user!
+  before_action :authenticate_request
+  after_action :hard_logout
+
   def index
-    @posts = Post.all.order('created_at')
+    @posts = User.find(params[:user_id]).posts.order('created_at')
```
Comment (Davidosky007) [inline app/controllers/API/V1/posts_controller.rb:8]: * Even after adding a bearer token I still get the error reported by the previous CR kindly look into it 
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @Graciano1997 @sheezu900 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
* Well-written PR 👍 
* No linter errors 👍 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidosky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by V-Blaze — APPROVED** [review]
Hi @kifle23 , 

# Approved 🟢 🟢 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy

## Highlights
- All required Endpoints created✔✔
- Followed a common practice to add  `api/v1` :heavy_check_mark: 
- All controller methods works as expected :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**